core/domain/stats_services.py

```diff
@@ -305,8 +305,12 @@ def update_exp_issues_for_new_exp_version(
     # Handling reverts.
     if revert_to_version:
         old_exp_issues = get_exp_issues(exploration.id, revert_to_version)
-        exp_issues.unresolved_issues = old_exp_issues.unresolved_issues
-        exp_issues.exp_version += 1
+        # If the old exploration issues model doesn't exist, the current model
+        # is carried over (this is a fallback case for some tests, and can
+        # never happen in production.)
```

Comment:
Hmm, I'd rather see the tests *also* avoid this case than to add a branch just to satisfy them. What can we do to make sure the tests never gets here (even if it means more complication)?

Reply:
@brianrodri There are a whole bunch of tests in exp_services_test, that do not use the _save_exploration() or the _create_exploration() methods in exp services. They explicitly do exp_models.ExplorationModel.create() and exp_models.ExplorationModel.commit(). Changing all these tests would rather be cumbersome, and I think some of them are meant to be that way (as mentioned in their docstrings). 

Having this check just provides an additional layer of security if you think about it. In normal cases, we are supposed to have ExplorationIssuesModel for all versions of an exploration leading up to the current version, but if there is some one off explicit case where we don't, this handles it.